### PR TITLE
[ie/Hotstar] year added in hotstar metadata

### PR DESCRIPTION
**Pull request by Rajeshwaran2001**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

HotStar Year MetaData Added

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
:sparkles::movie_camera::india:

Add `year` metadata field to HotStar extractor. This improves the accuracy and completeness of the video information extracted from the HotStar API.

> _`year` field added_
> _from `video_data` dict_
> _autumn of metadata_

### Walkthrough
* Add `year` field to video metadata from `video_data` dictionary ([link](https://github.com/yt-dlp/yt-dlp/pull/7869/files?diff=unified&w=0#diff-0a6703759f63eb9f8d2a82b989320bcfddf8e3cd634bbc3b4e4c10e643f6ed9eR286))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
Please fill out the template checklist or else the PR cannot be accepted:
> ### Before submitting a *pull request* make sure you have:
> - [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
> - [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
> - [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

> ### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
> - [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
> - [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

> ### What is the purpose of your *pull request*?
> - [ ] Fix or improvement to an extractor (Make sure to add/update tests)
> - [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
> - [ ] Core bug fix/improvement
> - [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

**Comment by pukkandan** [comment]
And update tests

File: yt_dlp/extractor/hotstar.py [hunk 1]
```diff
@@ -283,6 +283,7 @@ def _real_extract(self, url):
         return {
             'id': video_id,
             'title': video_data.get('title'),
+            'year': video_data.get('year'),
```
Comment (bashonly) [inline yt_dlp/extractor/hotstar.py:286]:
```suggestion
            'release_year': int_or_none(video_data.get('year')),
```

Comment (Rajeshwaran2001) [reply yt_dlp/extractor/hotstar.py]: suggestion  updated

**Comment by bashonly** [comment]
Update the tests with the `release_year`

**Comment by Rajeshwaran2001** [comment]
are you talking about unittest? if it is it is done

**Comment by bashonly** [comment]
I am talking about these:

```py
    _TESTS = [{
        'url': 'https://www.hotstar.com/can-you-not-spread-rumours/1000076273',
        'info_dict': {
            'id': '1000076273',
            'ext': 'mp4',
            'title': 'Can You Not Spread Rumours?',
            'description': 'md5:c957d8868e9bc793ccb813691cc4c434',
            'timestamp': 1447248600,
            'upload_date': '20151111',
            'duration': 381,
            'episode': 'Can You Not Spread Rumours?',
        },
        'params': {'skip_download': 'm3u8'},
    }, {
        'url': 'https://www.hotstar.com/tv/ek-bhram-sarvagun-sampanna/s-2116/janhvi-targets-suman/1000234847',
        'info_dict': {
            'id': '1000234847',
            'ext': 'mp4',
            'title': 'Janhvi Targets Suman',
            'description': 'md5:78a85509348910bd1ca31be898c5796b',
            'timestamp': 1556670600,
            'upload_date': '20190501',
            'duration': 1219,
            'channel': 'StarPlus',
            'channel_id': 3,
            'series': 'Ek Bhram - Sarvagun Sampanna',
            'season': 'Chapter 1',
            'season_number': 1,
            'season_id': 6771,
            'episode': 'Janhvi Targets Suman',
            'episode_number': 8,
        }
    }, {
```

**Comment by Rajeshwaran2001** [comment]
All Done and working fine

**Comment by Rajeshwaran2001** [comment]
Result of Test1: image
Result of Test2:image
Result of Test3:image
Result of Test4:image

Test 4 failed first due to broken link and i have update the test with working link and all test ok

**Comment by bashonly** [comment]
Why do none of the tests return a `release_year` field, though? The point of adding/updating tests is so we can test that `release_year` is succesfully extracted

**Comment by Rajeshwaran2001** [comment]
> Why do none of the tests return a `release_year` field, though? The point of adding/updating tests is so we can test that `release_year` is successfully extracted

`release_year` is mainly for movies and There is no test for that. You can see `release_year` extracted for a movie in the info.json file

image.

I can write a new test for this if you want.

**Comment by bashonly** [comment]
Yes please do

**Comment by Rajeshwaran2001** [comment]
image

A new test was added and verified

**Review by bashonly — APPROVED** [review]

**Pull request merged**